xarray/tests/test_dataarray.py

```diff
@@ -2961,9 +2963,11 @@ def test_resample(self):
         actual = array.resample(time="24H").reduce(np.mean)
         assert_identical(expected, actual)
 
+        # Our use of `loffset` may change if we align our API with pandas' changes.
+        # ref https://github.com/pydata/xarray/pull/4537
         actual = array.resample(time="24H", loffset="-12H").mean()
-        expected = DataArray(array.to_series().resample("24H", loffset="-12H").mean())
-        assert_identical(expected, actual)
+        expected = array.to_series().resample("24H").mean().index + to_offset("-12H")
+        assert_index_equal(actual, expected)
```

Comment:
to fix the tests, we probably need something like:
```suggestion
        expected_ = array.to_series().resample("24H").mean()
        expected_.index += to_offset("-12H")
        expected = DataArray.from_series(expected_)
        assert_identical(actual, expected)
```

Reply:
Thanks @keewis